src/plugins/data_source/server/plugin.ts

```diff
@@ -63,17 +71,48 @@ export class DataSourcePlugin implements Plugin<DataSourcePluginSetup, DataSourc
 
     const dataSourceService: DataSourceServiceSetup = await this.dataSourceService.setup(config);
 
+    core.logging.configure(
+      this.config$.pipe<LoggerContextConfigInput>(
+        map((dataSourceConfig) => ({
+          appenders: {
+            auditTrailAppender: dataSourceConfig.audit.appender,
+          },
+          loggers: [
+            {
+              context: 'audit',
+              level: 'info',
+              appenders: ['auditTrailAppender'],
+            },
+          ],
+        }))
+      )
+    );
+
+    const auditorFactory: AuditorFactory = {
+      asScoped: (request: OpenSearchDashboardsRequest) => {
+        return new LoggingAuditor(request, this.logger.get('audit'));
+      },
+    };
+    core.auditTrail.register(auditorFactory);
+    const auditTrailPromise = core.getStartServices().then(([coreStart]) => coreStart.auditTrail);
+
     // Register data source plugin context to route handler context
     core.http.registerRouteHandlerContext(
       'dataSource',
-      this.createDataSourceRouteHandlerContext(dataSourceService, cryptographyClient, this.logger)
+      this.createDataSourceRouteHandlerContext(
+        dataSourceService,
+        cryptographyClient,
+        this.logger,
+        auditTrailPromise
+      )
     );
 
     return {};
   }
 
   public start(core: CoreStart) {
     this.logger.debug('data_source: Started');
```

Comment:
For non-audit loggers, I wonder by `core.logging.configure()`, if it will also be sent to the file, for example this debug log. We obviously don't want that in the log file.

Reply:
it will not be there, noted that we specified context to be audit in logger